### DISCUSSION TITLE
Compilation error: 'class WiFiClientSecure' has no member named 'setInsecure'

### DISCUSSION THREAD

**Discussion by luc-github** [luc-github/ESP3DLib#26] [Q&A]
ESP32 core had new release and introduced new features, but marlin fixed version to old one and WebSocket library follow latest ESP32 core features
Solutions in platformIO.ini
```
[env:esp32]
platform = espressif32@1.11.2
#platform = https://github.com/platformio/platform-espressif32.git
board = esp32dev
build_flags = ${common.build_flags} -DCORE_DEBUG_LEVEL=0
lib_deps = ${common.lib_deps}
AsyncTCP=https://github.com/me-no-dev/AsyncTCP/archive/master.zip
ESPAsyncWebServer=https://github.com/.../ESPAsyncWebServer/archive/master.zip
ESP3DLib=https://github.com/luc-github/ESP3DLib.git
arduinoWebSockets=https://github.com/Links2004/arduinoWebSockets.git
```
1 - use previous version of websocket (add #2.3.4)
`arduinoWebSockets=https://github.com/Links2004/arduinoWebSockets.git#2.3.4`
another solution is to use latest esp32 core (remove fixed version)
2 - `platform = espressif32`

I did not tested solution 2 yet just solution 1, but it should work, I will update later
Edit: solution 2 is not working due to another issue  https://github.com/espressif/arduino-esp32/issues/4905

**Comment by Koxx3** [comment]
same issue for me.
doesn't build with IDF+ARDUINO for the same reason

**Comment by luc-github** [comment]
Marlin is supposed doing compilation using platformio not arduino ide, it may work but not using default settings for sure.
using arduino ide need specific esp32 core version and specific libraries version
you can try the https://github.com/luc-github/ESP3D installation but it was not tested as Marlin use platformio
